chirp/drivers/baofeng_uv17Pro.py

```diff
@@ -0,0 +1,1039 @@
+    # UI progress
+    status = chirp_common.Status()
+    status.cur = 0
+    status.max = radio.MEM_TOTAL // radio.BLOCK_SIZE
+    status.msg = "Cloning from radio..."
+    radio.status_fn(status)
+
+    for i in range(len(radio.MEM_SIZES)):
+        MEM_SIZE = radio.MEM_SIZES[i]
+        MEM_START = radio.MEM_STARTS[i]
+        for addr in range(MEM_START, MEM_START + MEM_SIZE, radio.BLOCK_SIZE):
+            frame = _make_read_frame(addr, radio.BLOCK_SIZE)
+            # DEBUG
+            LOG.debug("Frame=" + util.hexprint(frame))
+
+            # Sending the read request
+            baofeng_common._rawsend(radio, frame)
+
+            # Now we read data
+            d = baofeng_common._rawrecv(radio, radio.BLOCK_SIZE + 4)
+
+            LOG.debug("Response Data= " + util.hexprint(d))
+            d = _crypt(1, d[4:])
+
+            # Aggregate the data
+            data += d
+
+            # UI Update
+            status.cur = len(data) // radio.BLOCK_SIZE
+            status.msg = "Cloning from radio..."
+            radio.status_fn(status)
+    return data
+
+
+def _upload(radio):
+    # Put radio in program mode and identify it
+    _do_ident(radio)
+    for index in range(len(radio._magics)):
+        baofeng_common._rawsend(radio, radio._magics[index])
+        baofeng_common._rawrecv(radio, radio._magicResponseLengths[index])
+
+    data = b""
+
+    # UI progress
+    status = chirp_common.Status()
+    status.cur = 0
+    status.max = radio.MEM_TOTAL // radio.BLOCK_SIZE
+    status.msg = "Cloning from radio..."
+    radio.status_fn(status)
+
+    data_addr = 0x00
+    radio_mem = radio.get_mmap()
+    for i in range(len(radio.MEM_SIZES)):
+        MEM_SIZE = radio.MEM_SIZES[i]
+        MEM_START = radio.MEM_STARTS[i]
+        for addr in range(MEM_START, MEM_START + MEM_SIZE, radio.BLOCK_SIZE):
+            data = radio_mem[data_addr:data_addr + radio.BLOCK_SIZE]
+            data = _crypt(1, data)
+            data_addr += radio.BLOCK_SIZE
+
+            frame = _make_frame(b"W", addr, radio.BLOCK_SIZE, data)
+            # DEBUG
+            LOG.debug("Frame=" + util.hexprint(frame))
+
+            # Sending the read request
+            baofeng_common._rawsend(radio, frame)
+
+            # receiving the response
+            ack = baofeng_common._rawrecv(radio, 1)
+            if ack != b"\x06":
+                msg = "Bad ack writing block 0x%04x" % addr
+                raise errors.RadioError(msg)
+
+            # UI Update
+            status.cur = data_addr // radio.BLOCK_SIZE
+            status.msg = "Cloning to radio..."
+            radio.status_fn(status)
+    return data
+
+
+@directory.register
+class UV17Pro(baofeng_common.BaofengCommonHT):
+    """Baofeng UV-17Pro"""
+    VENDOR = "Baofeng"
+    MODEL = "UV-17Pro"
+    NEEDS_COMPAT_SERIAL = False
+
+    MEM_STARTS = [0x0000, 0x9000, 0xA000, 0xD000]
+    MEM_SIZES = [0x8040, 0x0040, 0x02C0, 0x0040]
+
+    MEM_TOTAL = 0x8380
+    BLOCK_SIZE = 0x40
+    STIMEOUT = 2
```

Comment:
Not used?

Reply:
Removed STIMEOUT. The other stuff is used